servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/DefaultSingleAddressHttpClientBuilder.java

```diff
@@ -552,4 +602,52 @@ public Completable closeAsyncGracefully() {
             return closeable.closeAsyncGracefully();
         }
     }
+
+    private static final class SdStatusCompletable extends Completable implements CompletableSource.Processor {
+        private static final AtomicReferenceFieldUpdater<SdStatusCompletable, CompletableSource.Processor>
+                currentProcessorUpdater = AtomicReferenceFieldUpdater.newUpdater(SdStatusCompletable.class,
+                CompletableSource.Processor.class, "currentProcessor");
+        @Nullable
+        private volatile CompletableSource.Processor currentProcessor;
+
+        @Override
+        protected void handleSubscribe(final CompletableSource.Subscriber subscriber) {
+            final CompletableSource.Processor currentProcessor = currentProcessorUpdater.updateAndGet(this,
```

Comment:
This approach is incorrect as it misses errors when `nextError()` happens first and then never happens again.

Instead what you need is to always have a `Processor` which queues the error till a subscriber arrives.

Once the error is queued, it needs a callback to clear the queued error once a successful event happens on the SD